include/bluetooth/bluetooth.h

```diff
@@ -404,6 +537,141 @@ int bt_le_adv_update_data(const struct bt_data *ad, size_t ad_len,
  */
 int bt_le_adv_stop(void);
 
+/** @brief Create advertising set.
+ *
+ *  Create a new advertising set and set advertising parameters.
+ *
+ *  @param[in] param Advertising parameters.
+ *  @param[in] cb    Callback struct to notify about advertiser activity. Can be
+ *                   NULL. Must point to valid memory during the lifetime of the
+ *                   advertising set.
+ *  @param[out] adv  Valid advertising set object on success or NULL otherwise.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_ext_adv_create(const struct bt_le_adv_param *param,
+			 const struct bt_le_ext_adv_cb *cb,
+			 struct bt_le_ext_adv **adv);
+
+struct bt_le_ext_adv_start_param {
+	/** Advertiser timeout (N * 10 ms), Application will be notified by the
+	 *  advertiser sent callback. Set to zero for no timeout.
+	 *
+	 *  If privacy :option:`CONFIG_BT_PRIVACY` is enabled then the timeout
+	 *  must be less than :option:`CONFIG_BT_RPA_TIMEOUT`.
+	 */
+	u16_t timeout;
+	/** Number of advertising events. Application will be notified by the
+	 *  advertiser sent callback. If used together with timeout then
+	 *  the condition that happens first will stop the advertiser.
+	 *  Set to zero for no limit.
+	 */
+	u8_t  num_events;
+};
+
+/** Start advertising with the given advertising set
+ *
+ *  Start advertising with a specific advertising set.
+ *  If both timeout and advertising event limit is given then the advertising
+ *  set will be stopped once one of them is reached.
+ *
+ *  @param adv    Advertising set object.
+ *  @param param  Advertise start parameters.
+ */
+int bt_le_ext_adv_start(struct bt_le_ext_adv *adv,
+			struct bt_le_ext_adv_start_param *param);
+
+/** Stop advertising with the given advertising set
+ *
+ *  Stop advertising with a specific advertising set. When using this function
+ *  the advertising sent callback will not be called.
+ *
+ *  @param adv Advertising set object.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_ext_adv_stop(struct bt_le_ext_adv *adv);
+
+/** Set advertising set advertising or scan response data.
+ *
+ *  Set advertisement data or scan response data. If the advertising set is
```

Comment:
Controllers don't support updating advertising data while advertising if doing extended advertising and the new length is more than 251 bytes. I assume that is also the intention here?

Reply:
There are some other similar requirements for not allowing extended scannable advertising without scan response data. It might therefore be sufficient to add a reference to the HCI spec

Reply:
I have added a note about the 251 bytes limit of updating while advertiser is active. 
Also added note to the SCANNABLE option that scan response data must be set.